### PR TITLE
Documentation (Best Practices for C-Plus-PLus) for HakinCodes

### PR DESCRIPTION
**Pull request by Kritika37**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description
Documentation (Best Practices for C-Plus-PLus)  including naming style and indentation

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [X] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#544

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Kritika37 please use Grammarly and check `.md` file since there is lot of typos mistakes and update the same. Also, format the markdown file using markdown formatter since there is lot of line break at the end of the file.

**Comment by Kritika37** [comment]
> @Kritika37 please use Grammarly and check `.md` file since there is lot of typos mistakes and update the same. Also, format the markdown file using markdown formatter since there is lot of line break at the end of the file.

Ok

File: C-Plus-Plus/BEST_PRACTICES_C-PLUS-PLUS.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+   MyClass
+   ```
+  
+4. All other Parameters use snake case 
+   
+   ```
+   with_map
+   ```
+
+## Whitespace and Identation
```
Comment (iamrajiv) [inline C-Plus-Plus/BEST_PRACTICES_C-PLUS-PLUS.md]: @Kritika37 fix typo of `Identation`

File: C-Plus-Plus/BEST_PRACTICES_C-PLUS-PLUS.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+   * The most important thing is never mix tabs and spaces.
+   * Use 2 or 4 spaces per level of identation.
```
Comment (iamrajiv) [inline C-Plus-Plus/BEST_PRACTICES_C-PLUS-PLUS.md]: this too

File: C-Plus-Plus/BEST_PRACTICES_C-PLUS-PLUS.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+   * You Should set your editor to replace tabs with spaces.
+   * Add space to improve readability between operators and operands,commas and semicolon.
+   * Add space on outside of parantheses and brackets wherever needed.
```
Comment (iamrajiv) [inline C-Plus-Plus/BEST_PRACTICES_C-PLUS-PLUS.md]: fix typo of `parantheses`

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/BEST_PRACTICES_C-PLUS-PLUS.md (hunk 3 above)
Comment (Kritika37) [reply C-Plus-Plus/BEST_PRACTICES_C-PLUS-PLUS.md]: yes I'm working on it

**Pull request closed without merging**